tests/foreman/ui_airgun/test_syncplan.py

```diff
@@ -473,3 +523,111 @@ def test_positive_synchronize_custom_product_weekly_recurrence(
         repo_values = session.repository.read(product.name, repo.name)
         for repo_type in ['Packages', 'Errata', 'Package Groups']:
             assert repo_values['content_counts'][repo_type] > '0'
+
+
+@tier3
+def test_positive_synchronize_custom_product_custom_cron_real_time(
+        session, module_org):
+    """Create a sync plan with past datetime as a sync date,
+    add a custom product and verify the product gets synchronized
+    on the next sync occurrence based on custom cron interval
+
+    :id: 4d9ed0bf-a63c-44de-846d-7cf302273bcc
+
+    :expectedresults: Product is synchronized successfully.
+
+    :BZ: 1279539
+
+    :CaseLevel: System
+    """
+    plan_name = gen_string('alpha')
+    product = entities.Product(organization=module_org).create()
+    repo = entities.Repository(product=product).create()
+    with session:
+        startdate = session.browser.get_client_datetime()
+        # calculating next sync based on (*/5 * * * *) will have occurrences (0, 5, 10, 15..)
+        next_sync = 5 - (int(startdate.strftime('%M')) % 5)
```

Comment:
there is a validation where we have to check sync plan yet not started. This happens to be going to change based on the current minute time  
e.g. 
1. if the current time is  21:03 -> as the minute is 03 and cron expression is "*/5 * * * *" then the next syn will be 21:05, as mentioned as delay won't work be to see if the task in pending status. instead of that, I thought good check current minute difference with the next 5 minutes and based on that select the delay and next sync occurrence.

I hope this clears my intention. If you have a better solution let me know.

Reply:
@oshtaier : Are you ok with my explanation, if not please let me know

Reply:
then `time.sleep(3/2)`. hmm. there is a chance that you didn't catch situation when it will fail, so maybe we still need to operate with seconds, not minutes.
But I got your idea and it looks very good

Reply:
and sorry for delay with response. It is too many PRs and I miss that one by accident

Reply:
> is too many PRs and I miss that one by

I can understand that :)

Reply:
> Actually:
> 
> ```
> def sleep(seconds):
>     """
>     sleep(seconds)
>     
>     Delay execution for a given number of seconds.  The argument may be
>     a floating point number for subsecond precision.
>     """
>     pass
> ```
> so how your code is suppose to work?

Sorry I did not understood your intension here

Reply:
understood, need to * by 60 got you!